framework/lib/components/toolbox/toolbox.tsx

```diff
@@ -62,23 +64,68 @@ export const Toolbox = ({
     }
   }
 
-  const toolboxWidth = (width ?? container.w) as string
+  const getPixelSize = (sizeToken: string) => {
+    const widthInRem = useToken('sizes', sizeToken)
+    return parseFloat(widthInRem) * 16
+  }
+
+  const defaultWidth = getPixelSize(size)
+  const minWidth = getPixelSize('sm')
+  const maxWidth = getPixelSize('2xl')
+
+  const [ adjustableWidth, setAdjustableWidth ] = useState(defaultWidth)
 
   return (
-    <Box w={ toolboxWidth } display={ isOpen && shouldPush ? 'initial' : 'none' }>
+    <Box w={ adjustableWidth } display={ isOpen && shouldPush ? 'initial' : 'none' }>
       <Portal>
         <Slide
           direction={ direction }
           in={ isOpen }
           style={ {
             height: container.h as string,
-            width: toolboxWidth,
+            width: adjustableWidth,
             zIndex: coreZIndex.overlay,
           } }
         >
-          <Flex sx={ { ...container, w: toolboxWidth } } { ...rest } onKeyDown={ handleKeyDown }>
+          <Flex
+            sx={ { ...container, w: adjustableWidth } }
+            { ...rest }
```

Comment:
This prevents rest properties from overriding `onKeyDown` and `style` props.
Please move `{...rest}` to be the last on the props list.